### DISCUSSION TITLE
Rename mojo please

### DISCUSSION THREAD

**Discussion by stefanuytterhoeven** [modular/modular#2077] [General]
I know. It won't happen I guess. But I try to search for info about mojo .. and the search term needs always to contain 'mojo language' .  There are so many search results containing 'mojo' that has nothing to do with modular mojo. Even frameworks...

**Comment by ksandvik** [comment]
We have to ensure that mojo is so successful it ends up on the top web searches.

**Comment by Sharktheone** [comment]
I guess, Rust, Go, D, R and C also don't have it easier 😄

**Comment by rarebreed** [comment]
Yeah, lots of languages have that problem.  If I go on someone else's computer and search for python, I'm not going to get anything about the language.  Ditto for Julia, Dart, as well as the ones Sharktheone mentioned.

Once the browser starts tracking you though, and learns your search history, it will give you better results.  Unless you're going hardcore bravo or tor mode, I wouldn't sweat it that much.

That being said, I think mojo **does** need a name for its mascot. 

- match?
- sparky?
- fireball?
- torch?

I also wonder what mojo has to do with a flame (yeah,...I know....the emoji).  I mean, if mojo is supposed to be about magic, why not use the mage emoji?🧙‍♂️  Or maybe the magic wand?🪄

I'd actually petition for a change in mascot and emoji prefix if the name `mojo` is kept

**Reply by debangshu-mukherjee** [reply]
> Yeah, lots of languages have that problem. If I go on someone else's computer and search for python, I'm not going to get anything about the language. Ditto for Julia, Dart, as well as the ones Sharktheone mentioned.

Yes. A few years back Nicki Minaj's song used to come as the top result when searching for Anaconda

**Comment by geo-mak** [comment]
I think mojo is not a serious name anyway! It is considered derogatory slang by many individuals.
Make it "Fire" or "Dragon" (fits the LLVM branding!), since it is advertised as Python with "fire"!

**Reply by geo-mak** [reply]
I have admiration for the work and I genuinely appreciate everyone involved. This fusion of Python with elements I like in Rust (my choice for addressing Python's performance issues) is a dream come true, I just can't relate to the name 😅

**Reply by stefanuytterhoeven** [reply]
I think, the more unique the name is, the easier you can search for info 
And since it's a new language, a lot of searching is needed.

**Reply by geo-mak** [reply]
I'm not particularly concerned about uniqueness because you can likely find another usage for it on the internet anyway, I feel that the name 'mojo' is somewhat esoteric and may convey non-obvious meanings!

**Reply by rarebreed** [reply]
I'm curious how it is derogatory?

Mojo is supposed to mean something magical, or at least that was the original meaning.  It'd be a shame if the meaning has been subverted and reappropriated...which sadly, is happening a lot nowadays.

I do kind of agree that something related to snakes would be more appropriate since it's supposed to be a superset of Python.  Or actually, since python was _originally_ named after the English comedy troupe, something related to _Monty Python_ would also be appropriate.

**Reply by geo-mak** [reply]
I'm not explicitly stating that it's derogatory, but the meaning isn't obvious enough for something aiming for popularity.

**Comment by rarebreed** [comment]
Since python was originally named after _Monty Python_ (not the snake), I think something in the spirit of _Monty Python_ would also be acceptable.  

So how about we name it `gambolputty`?  After that lost and forgotten German Baroque music composer who should have been up there with the other greats like Bach and Handel, whose name has been lost to history, Johann Gambolputty...

[...de von Ausfern-schplenden-schlitter-crasscrenbon-fried-digger-dingle-dangle-dongle-dungle-burstein-von-knacker-thrasher-apple-banger-horowitz-ticolensic-grander-knotty-spelltinkle-grandlich-grumblemeyer-spelterwasser-kurstlich-himbleeisen-bahnwagen-gutenabend-bitte-ein-nürnburger-bratwustle-gerspurten-mitzweimache-luber-hundsfut-gumberaber-shönendanker-kalbsfleisch-mittler-aucher von Hautkopft of Ulm](https://montypython.fandom.com/wiki/Johann_Gambolputty_(character))

**Comment by iamtimdavis** [comment]
Thanks for all the discussion, but we won't be changing the name at all :) You can refer to why we named it Mojo 🔥 here - https://docs.modular.com/mojo/faq#why-is-it-called-mojo

Over time, as with all languages (and technology products), we strongly believe Mojo🔥 will stand on its own and have its own clear identity. I'd encourage you to remember, it's only been around 11 months since we first announced it. We definitely need yours (and the entire communities) help to create more Mojo🔥 code and incredible projects, publish more articles and tutorials, contribute to helping us grow, improve, build and deliver Mojo🔥 to the world. 

We see this as a long term effort, and we are deeply committed to it - we hope you are all too! 🚀

**Reply by stefanuytterhoeven** [reply]
https://www.google.com/search?q=mojo&oq=mojo+&gs_lcrp=EgZjaHJvbWUyBggAEEUYOTIGCAEQRRg8MgYIAhBFGDwyBggDEEUYPDIGCAQQRRg8MgYIBRBFGDwyBggGEEUYPDIGCAcQRRg7MgYICBBFGDsyBggJEEUYO9IBCDI2NzBqMGo0qAIBsAIB&client=ms-android-samsung-ss&sourceid=chrome-mobile&ie=UTF-8&chrome_dse_attribution=1
Look for mojo... Prepare to scroll down A LOT.
